modules/rosetta/src/test/scala/org/tessellation/rosetta/domain/api/construction/ConstructionPayloadsSuite.scala

```diff
@@ -0,0 +1,150 @@
+package org.tessellation.rosetta.domain.api.construction
+
+import cats.effect.IO
+import cats.syntax.either._
+
+import org.tessellation.rosetta.domain.api.construction.ConstructionPayloads.Response
+import org.tessellation.rosetta.domain.generators._
+
+import io.circe.Json
+import io.circe.parser.{decode, parse}
+import io.circe.syntax.EncoderOps
+import org.scalacheck.Gen
+import weaver.SimpleMutableIOSuite
+import weaver.scalacheck.Checkers
+
+object ConstructionPayloadsSuite extends SimpleMutableIOSuite with Checkers {
+
+  val responseGen: Gen[Response] = for {
+    unsignedTransaction <- hexGen
+    payloads <- payloadGen // for simplicity just make one payload
+  } yield Response(unsignedTransaction, payloads :: Nil)
+
+  test("decode sample request") { _ =>
+    // Sample taken from https://www.rosetta-api.org/docs/models/ConstructionPayloadsRequest.html
+    // with following changes:
+    //  blockchain "bitcoin" to "dag"
+    //  status     "Reverted" to "Accepted"
+    //  address    changed all addresses to a valid DAGAddress
+    //  symbol     "BTC" to "DAG"
+    val sampleJson =
```

Comment:
it doesn't look like a unit test. I'd recommend to put all json-schema tests to separate directory or even a separate test module.
anyway, for unit testing the api, check out the HttpSpec (expectBody, expectStatus, etc) - see the test I've already prepared for rosetta

Reply:
@marcinwadon I haven't found `HttpSpec` nor have I found the rosetta test you mentioned. I do see `test-shared:HttpSuite` -- is that what you mean by putting them in a separate directory?

Reply:
first thing is that it's more like an integration test. (if it cannot be done via rosetta tool, I'd suggest moving it to `it` test module). and yes, I meant the `HttpSuite` not `HttpSpec`, to take a look how you can unit test specific endpoints

Reply:
Removing -- I will revisit this test in another PR.